src/test/java/seedu/address/logic/commands/project/EditTaskCommandTest.java

```diff
@@ -0,0 +1,181 @@
+        expectedModel.getFilteredProjectList().get(INDEX_FIRST_PROJECT.getZeroBased())
+                .deleteTask(project.getFilteredSortedTaskList().get(0));
+        expectedModel.getFilteredProjectList().get(INDEX_FIRST_PROJECT.getZeroBased())
+                .addTask(editedTask);
+
+        assertCommandSuccess(editTaskCommand, model, expectedMessage, expectedModel);
+
+    }
+
+    @Test
+    public void execute_noFieldSpecifiedUnfilteredList_success() throws ParseException {
+
+        Project project = model.getFilteredProjectList().get(INDEX_FIRST_PROJECT.getZeroBased());
+        model.enter(project);
+        ModelManager expectedModel = new ModelManager(model.getProjectCatalogue(), new UserPrefs());
+        Task editedTask = project.getFilteredSortedTaskList().get(0);
+
+        EditTaskCommand editTaskCommand = new EditTaskCommand(ParserUtil.parseIndex("1"),
+                new EditTaskCommand.EditTaskDescriptor());
+
+        String expectedMessage = String.format(EditTaskCommand.MESSAGE_EDIT_TASK_SUCCESS, editedTask);
+
+        Project projectCopy = new Project(project.getProjectName(), project.getDeadline(),
+                project.getRepoUrl(), project.getProjectDescription(), project.getProjectTags(),
+                new HashMap<>(), project.getTasks());
+        expectedModel.setProject(project, projectCopy);
+        expectedModel.enter(projectCopy);
+
+        assertCommandFailure(editTaskCommand, model, EditTaskCommand.MESSAGE_NOT_EDITED);
+
+    }
+
+    @Test
+    public void execute_duplicateTaskFilteredList_failure() throws ParseException {
+        Project project = model.getFilteredProjectList().get(INDEX_FIRST_PROJECT.getZeroBased());
+        model.enter(project);
+        Task editedTask = project.getFilteredSortedTaskList().get(1);
+        EditTaskCommand.EditTaskDescriptor descriptor = new EditTaskDescriptorBuilder(editedTask).build();
+        EditTaskCommand editTaskCommand = new EditTaskCommand(ParserUtil.parseIndex("1"), descriptor);
+
+        assertCommandFailure(editTaskCommand, model, EditTaskCommand.MESSAGE_DUPLICATE_TASK);
+    }
+
+    @Test
+    public void execute_duplicateProjectFilteredList_failure() throws ParseException {
+        Project project = model.getFilteredProjectList().get(INDEX_FIRST_PROJECT.getZeroBased());
+        model.enter(project);
+        project.addTask(SampleDataUtil.generateTask(SampleDataUtil.getTask4()));
+        project.addTask(SampleDataUtil.generateTask(SampleDataUtil.getTask6()));
+        project.updateTaskFilter(x->x.getTaskName().contains("Write"));
+        Task editedTask = project.getFilteredSortedTaskList().get(0);
+        EditTaskCommand.EditTaskDescriptor descriptor = new EditTaskDescriptorBuilder(editedTask).build();
+        EditTaskCommand editTaskCommand = new EditTaskCommand(ParserUtil.parseIndex("2"), descriptor);
+
+        assertCommandFailure(editTaskCommand, model, EditTaskCommand.MESSAGE_DUPLICATE_TASK);
+    }
```

Comment:
Is this supposed to be in taskcommandtest?

Reply:
fixed thanks!